pkg/synthetictests/historicaldata/next_best_guess.go

```diff
@@ -0,0 +1,163 @@
+	MinorReleaseUpgrade,
+	PreviousReleaseUpgrade,
+	combine(PreviousReleaseUpgrade, MicroReleaseUpgrade),
+	combine(PreviousReleaseUpgrade, MinorReleaseUpgrade),
+	OnOVN,
+	OnSDN,
+	combine(PreviousReleaseUpgrade, OnOVN),
+	combine(PreviousReleaseUpgrade, OnSDN),
+	combine(PreviousReleaseUpgrade, MicroReleaseUpgrade, OnOVN),
+	combine(PreviousReleaseUpgrade, MicroReleaseUpgrade, OnSDN),
+	combine(PreviousReleaseUpgrade, MinorReleaseUpgrade, OnOVN),
+	combine(PreviousReleaseUpgrade, MinorReleaseUpgrade, OnSDN),
+}
+
+/*
+  {
+    "BackendName": "ingress-to-oauth-server-reused-connections",
+    "Release": "4.11",
+    "FromRelease": "4.10",
+    "Platform": "azure",
+    "Network": "ovn",
+    "Topology": "ha",
+    "P95": "3.8499999999999996",
+    "P99": "3.9699999999999998"
+  },
+  {
+    "BackendName": "ingress-to-oauth-server-reused-connections",
+    "Release": "4.11",
+    "FromRelease": "4.10",
+    "Platform": "azure",
+    "Network": "sdn",
+    "Topology": "ha",
+    "P95": "0.0",
+    "P99": "12.869999999999884"
+  },
+
+*/
+
+// NextBestKey returns the next best key and a bool indicating whether this guesser has an opinion.
+// If the bool is false, the key should not be used.
+type NextBestKey func(in platformidentification.JobType) (platformidentification.JobType, bool)
+
+// MinorReleaseUpgrade if we don't have data for the current fromRelease and it's a micro upgrade, perhaps we have data
+// for a minor upgrade.
+func MinorReleaseUpgrade(in platformidentification.JobType) (platformidentification.JobType, bool) {
+	if len(in.FromRelease) == 0 {
+		return platformidentification.JobType{}, false
+	}
+
+	fromReleaseMinor := getMinor(in.FromRelease)
+	toReleaseMajor := getMajor(in.Release)
+	toReleaseMinor := getMinor(in.Release)
+	// if we're already a minor upgrade, this doesn't apply
+	if fromReleaseMinor == (toReleaseMinor - 1) {
+		return platformidentification.JobType{}, false
+	}
+
+	ret := platformidentification.CloneJobType(in)
+	ret.FromRelease = fmt.Sprintf("%d.%d", toReleaseMajor, toReleaseMinor-1)
+	return ret, true
+}
+
+// MicroReleaseUpgrade if we don't have data for the current fromRelease and it's a minor upgrade, perhaps we have data
+// for a minor upgrade.
+func MicroReleaseUpgrade(in platformidentification.JobType) (platformidentification.JobType, bool) {
+	if len(in.FromRelease) == 0 {
+		return platformidentification.JobType{}, false
+	}
+
+	fromReleaseMinor := getMinor(in.FromRelease)
+	toReleaseMajor := getMajor(in.Release)
+	toReleaseMinor := getMinor(in.Release)
+	// if we're already a micro upgrade, this doesn't apply
+	if fromReleaseMinor == toReleaseMinor {
+		return platformidentification.JobType{}, false
+	}
+
+	ret := platformidentification.CloneJobType(in)
+	ret.FromRelease = fmt.Sprintf("%d.%d", toReleaseMajor, toReleaseMinor)
+	return ret, true
+}
+
+// PreviousReleaseUpgrade if we don't have data for the current toRelease, perhaps we have data for the congruent test
+// on the prior release.
```

Comment:
Example would be nice here in the godoc for each of these, i.e. x will fall back to y.